File: backend/app/schemas/roadmap.py
from pydantic import BaseModel


class StageResponse(BaseModel):
    id: int
    order_number: int
    title: str
    description: str
    unlock_quiz_score: int
    unlock_exercise_completion: int
    unlocked: bool = False


class RoleRoadmapResponse(BaseModel):
    role_id: int
    role_name: str
    stages: list[StageResponse]

File: src/types/roadmap.ts
export interface StageResponse {
  id: number;
  order_number: number;
  title: string;
  description: string;
  unlock_quiz_score: number;
  unlock_exercise_completion: number;
  unlocked: boolean;
  completed: boolean;
}

export interface RoleRoadmapResponse {
  role_id: number;
  role_name: string;
  stages: StageResponse[];
}
